### PR TITLE
Add `pylance-version` option

### PR DESCRIPTION
**Pull request by debonte**
Part of https://github.com/microsoft/pylance-release/issues/4927

Added a new `pylance-version` option which can be `latest-release`, `latest-prerelease`, or a specific Pylance version number like `2023.11.11`.

When `pylance-version` is provided and `version` is not, we'll leverage the Pylance release metadata JSON files in https://github.com/microsoft/pylance-release/tree/main/releases to determine which version of Pyright is included in that Pylance release, and then download and run that Pyright build as usual.

### CODE REVIEW

File: src/helpers.ts
Comment (jakebailey) [inline src/helpers.ts:250]: Can you make this a valita validator rather than casting? See the registry response above for an example.

Comment (debonte) [reply src/helpers.ts]: Ah, that's cool. Done.

File: src/helpers.ts
Comment (jakebailey) [inline src/helpers.ts:238]: I don't mind if this is how you want to do it, but would it make sense to have a single file with the mappings again? The directory will end up growing indefinitely which sounds like it'd get unweildy in time.

**Review by jakebailey — COMMENTED** [review]
Overall this seems good; you will need to also update the `README` as well to document the option, and maybe add an example.

File: src/helpers.ts
Comment (debonte) [reply src/helpers.ts:238]: I hear what you're saying, but I think the pain of a large directory will be minimal, if not zero. Having one file per version simplifies the code that generates and consumes these files. Since the consumers (initially pyright-action and pyright-python) are all expected to be external, I think that simplification outweighs the possible pain.

I don't expect people to be looking at these files much, but if they do, the one downside I see is that GitHub only displays the first 1000 files in a directory. In the past 12 months we shipped ~110 releases. We're considering switching to nightly prerelease and monthly stable releases, and then I'm guessing we'll ship around 250 releases per year. If the 1000 file display limit becomes an issue, I would be fine with deleting, say, all of the files that are more than 2-3 years old. If someone is pinned on a version that old, I'm ok with breaking them. They should upgrade.

Comment (jakebailey) [reply src/helpers.ts:238]: No worries; I just asked because once this is in, it's there for good.

**Review by jakebailey — APPROVED** [review]
This seems good to me; thanks for sending this. Any other remaining concerns? If not, I'll cut a release for this.

**Comment by debonte** [comment]
> Any other remaining concerns?

I should be sending a matching PR to pyright-python today. In the off-chance that Robert has any feedback that somehow impacts this change, would you mind holding off on merging this until he approves my PR?

**Comment by jakebailey** [comment]
I'm now out at a morale event so can't do it now anyway; just let me know when you're ready and I'll merge and release it!

**Comment by debonte** [comment]
FYI, I'm still waiting for feedback on my pyright-python PR. I'm guessing Robert is on vacation. He was quite responsive a couple weeks ago, but has been silent over the past week. I'll let you know if this can be merged as soon as I hear back from him on https://github.com/RobertCraigie/pyright-python/pull/230.

**Comment by debonte** [comment]
@jakebailey, the `pyright-python` PR was merged without any changes that would impact this PR. So please merge this at your convenience. Thanks!

**Comment by jakebailey** [comment]
Released in v1.8.0. Thanks!

**Pull request merged**
